presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractArrayElementBlock.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+    @Override
+    public <T> T getObject(int position, Class<T> clazz)
+    {
+        checkReadablePosition(position);
+        return getBlock().getObject(position + start, clazz);
+    }
+
+    @Override
+    public boolean bytesEqual(int position, int offset, Slice otherSlice, int otherOffset, int length)
+    {
+        checkReadablePosition(position);
+        return getBlock().bytesEqual(position + start, offset, otherSlice, otherOffset, length);
+    }
+
+    @Override
+    public int bytesCompare(int position, int offset, int length, Slice otherSlice, int otherOffset, int otherLength)
+    {
+        checkReadablePosition(position);
+        return getBlock().bytesCompare(position + start, offset, length, otherSlice, otherOffset, otherLength);
+    }
+
+    @Override
+    public void writeBytesTo(int position, int offset, int length, BlockBuilder blockBuilder)
+    {
+        checkReadablePosition(position);
+        getBlock().writeBytesTo(position + start, offset, length, blockBuilder);
+    }
+
+    @Override
+    public void writePositionTo(int position, BlockBuilder blockBuilder)
+    {
+        checkReadablePosition(position);
+        getBlock().writePositionTo(position + start, blockBuilder);
+    }
+
+    @Override
+    public boolean equals(int position, int offset, Block otherBlock, int otherPosition, int otherOffset, int length)
+    {
+        checkReadablePosition(position);
+        return getBlock().equals(position + start, offset, otherBlock, otherPosition, otherOffset, length);
+    }
+
+    @Override
+    public int hash(int position, int offset, int length)
+    {
+        checkReadablePosition(position);
+        return getBlock().hash(position + start, offset, length);
+    }
+
+    @Override
+    public int compareTo(int leftPosition, int leftOffset, int leftLength, Block rightBlock, int rightPosition, int rightOffset, int rightLength)
+    {
+        checkReadablePosition(leftPosition);
+        return getBlock().compareTo(leftPosition + start, leftOffset, leftLength, rightBlock, rightPosition, rightOffset, rightLength);
+    }
+
+    @Override
+    public Block getSingleValueBlock(int position)
+    {
+        return getRegion(position, 1, true);
+    }
+
+    @Override
+    public boolean isNull(int position)
+    {
+        checkReadablePosition(position);
+        return getBlock().isNull(position + start);
+    }
+
+    @Override
+    public void assureLoaded()
+    {
+    }
+
+    @Override
+    public BlockEncoding getEncoding()
+    {
+        // ArrayElementBlock does not exist
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Block getRegion(int position, int length)
+    {
+        return getRegion(position, length, false);
+    }
+
+    @Override
+    public Block copyRegion(int position, int length)
+    {
+        return getRegion(position, length, true);
+    }
+
+    private Block getRegion(int position, int length, boolean compact)
```

Comment:
inline this method, since it just throws and exception. You can remove the comment too